Recipes/Recipes.API/Controllers/RecipesController.cs

```diff
@@ -27,18 +28,35 @@ public class RecipesController : ControllerBase
     [HttpGet]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<IEnumerable<Recipe>>> GetAllRecipes()
+    public ActionResult<IEnumerable<Recipe>> GetAllRecipes([FromQuery] string? title, [FromQuery] string? csvTags, [FromQuery] long? authorId, [FromQuery] int? skip, [FromQuery] int? top)
     {
-        IEnumerable<RecipeEntity> recipes;
+        IQueryable<RecipeEntity> recipesQuery;
         try
         {
-            recipes = await _recipesRepository.GetAllAsync();
+            recipesQuery = _recipesRepository.GetQueryable();
```

Comment:
You just made the repository pattern useless :D

Reply:
I guess I got a little caried away putting everything in controllers. 
Moved filtering logic to Repository